backend/src/controllers/transaction-contorller.ts

```diff
@@ -0,0 +1,25 @@
+import { Request, Response } from "express";
+import { createTransaction } from "../repositories/transaction-repository";
+import {
+  getWalletByUserId,
+  updateWalletBalance,
+} from "../repositories/wallet-repositry";
+
+export const addCoins = async (req: Request, res: Response) => {
```

Comment:
what does adding coins mean in the controller? user can add coin? it's the name for like service. change naming to a better and readable name.